### DISCUSSION TITLE
RSSI equation

### DISCUSSION THREAD

**Discussion by achaitu** [NVlabs/sionna#411] [Q&A]
In Sionna RT, assume a scene with only reflections (no scattering, diffraction, refraction). For example, both Tx and Rx are on the same side of a wall. In this case, I want to know the equation to compute the RSSI at Rx given the scene at Tx give a max_depth. How is coverage_map giving me the final RSSI values? I'm looking at the source https://nvlabs.github.io/sionna/_modules/sionna/rt/coverage_map.html#CoverageMap but I could find an expression for the RSSI values except the `cm = 10.*np.log10(self[tx].numpy())` but how is the scene, paths, attenutations used to get the value? Where is the Monte Carlo estimate equation by shooting N rays?

I believe [equation 44](https://nvlabs.github.io/sionna/api/rt.html#equation-cm-mc-ref) in RT docs and [equation 26](https://nvlabs.github.io/sionna/em_primer.html#equation-h-final) may be useful. 

Also, can I get RSSI for each depth in max_depth? 

I want to see the expression for RSSI in summations/integrals in this setting. In other words, what all variables/constants do I need so that I can write an expression for RSSI at Rx? Please assume a very toy setting as the example above. You can point me to the correct equations or write out the equations. Thanks!

**Comment by jhoydis** [comment]
Hi @achaitu,

The coverage map describes the pathloss at a specific position. The received signal strength could be computed by assuming a specific transmit power, e.g., P_RX_dB = P_TX_dB + Pathloss_dB.

I am not sure how exactly RSSI relates to received signal power. This is probably standard dependent, e.g. WiFi.

As an alternative to coverage maps, you could also compute the channel impulse response at a particular position using `scene.compute_paths()` and then simply add the energy of the channel taps `a` together. This would lead to
P_RX = P_TX * sum_i |a_i|^2.

**Comment by souhila98** [comment]
Hello,
I am using Sionna and I would like to check whether my RSSI computation is correct.
I compute the RSSI in two different ways:
1. From the CFR , by computing the average received power over all Rx antennas, Tx antennas, and subcarriers, and then converting it to dB.
2. From the ray-tracing paths, by summing the powers of all received paths and then converting the result to dB.

With the first method, using the CFR, I observe stronger fading fluctuations. With the second method, using the path amplitudes, the RSSI looks smoother.

Could you please tell me if this behavior is expected, or if there might be an error in the way I compute one of these two quantities?

Thank you.

<img width="650" height="254" alt="cfr_path" src="https://github.com/user-attachments/assets/595ea89b-848a-4f12-9210-1d9ff8b4b181" />

**Reply by merlinND** [reply]
Hello @souhila98,

I believe that this is expected for coverage maps, see https://github.com/NVlabs/sionna/discussions/645.
